lib/checkr.rb

```diff
@@ -61,6 +61,8 @@ def self.api_url(path='')
   end
 
   def self.request(method, path, params={}, headers={})
+    api_key = headers.key?(:api_key) ? headers.delete(:api_key) : @api_key
```

Comment:
why do we need to delete the key?

Reply:
I'd recommend to set this as different param

Reply:
Sorry I caused confusion with the usage of 'param' here. Ideally I wanted to see it in another argument. Given complexity for this, and that's it's been implemented like this (using headers) in other public gem, I'm ok with it.